### DISCUSSION TITLE
Run shell script on window close via --watcher

### DISCUSSION THREAD

**Discussion by Strykar** [kovidgoyal/kitty#3311] [Q&A] [answered]
kitty is started with `--watcher=/home/strykar/.config/kitty/foo_watcher.py` and the machine has `Python 3.9.1`

The shell script just starts the program running in the same tab and location as it was before it was closed for whatever reason:
`kitty @ launch --match title:foo --location first --keep-focus bash -c "/usr/bin/nice -10 foo"`

I don't know python and tried to make a watcher python wrapper to call this script when the process dies, as the docs appear to suggest to me... but killing the program (which closes that particular window in the tab), does not trigger the shell script, what am I missing?
```
$ cat .config/kitty/foo_watcher.py 
import subprocess
from kitty.boss import Boss
from kitty.window import Window


def on_close(boss, window, data):
    rc = subprocess.call("foo.sh")
```

**Comment by kovidgoyal** [comment, marked as answer]
First you dont need a shell script, just call

boss.new_window('bash',  '-c', 'foo')

secondly, before doing that make sure the watcher is properly working by writing to a log file in on_close() like this

open('/tmp/test', 'w').write('working')
